lexos/static/js/scripts_cut.js

```diff
@@ -1,382 +1,663 @@
+    formData.append('action', action)
+    const jsonform = jsonifyForm()
+    $.extend(jsonform, {'action': action})
+    // Initiate a timer to allow user to cancel if processing takes too long
+    const loadingTimeout = window.setTimeout(function () {
+        $('#needsWarning').val('true')
+        const timeWarning = 'Lexos seems to be taking a long time. This may \
+        be because you are cutting a large number of documents. \
+        If not, we suggest that you cancel, reload the page, and try again.'
+        const footerButtons = '<button type="button" class="btn btn-default"\
+        data-dismiss="modal">Continue Anyway</button>\
+        <button type="button" class="btn btn-default" id="timerCancel" >\
+        Cancel</button>'
+        $('#warning-modal-footer').html(footerButtons)
+        $('#warning-modal-message').html(timeWarning)
+        $('#warning-modal').modal()
+    }, 10000) // 10 seconds
+    xhr = $.ajax({
+        url: '/doCutting',
+        type: 'POST',
+        processData: false, // important
+        contentType: false,
+        data: formData,
+        error: function (jqXHR, textStatus, errorThrown) {
+            $('#status-prepare').css({'visibility': 'hidden'})
+            // Show an error if the user has not cancelled the action
+            if (errorThrown !== 'abort') {
+                const cull_msg = 'Lexos could not apply the cutting actions.'
+                $('#error-modal-message').html(cull_msg)
+                $('#error-modal').modal()
+            }
+            console.log('bad: ' + textStatus + ': ' + errorThrown)
+        }
+    }).done(function (response) {
+        clearTimeout(loadingTimeout)
+        $('#warning-modal').modal('hide') // Hide the warning if it is displayed
+        response = JSON.parse(response)
+        $('#preview-body').empty() // Correct
+        j = 0
+        $.each(response['data'], function () {
+            let fileID = $(this)[0]
+            let filename = $(this)[1]
+            let fileLabel = filename
+            let fileContents = $(this)[3]
+            const indivcutbuttons = '<a id="indivcutbuttons_${fileID}" \
+            onclick="toggleIndivCutOptions(${fileID});" \
+            class="bttn indivcutbuttons" \
+            role="button">Individual Options</a></legend>'
+
+            // CSS truncates the document label
+            const fieldsetSelector = '<fieldset \
+            class="individualpreviewwrapper"><legend \
+            class="individualpreviewlegend has-tooltip" \
+            style="color:#999; width:90%;margin: auto; white-space: nowrap; \
+            overflow: hidden; text-overflow: ellipsis;">\
+            ' + fileLabel + ' ' + indivcutbuttons + '</fieldset>'
+
+            let fieldset = $(fieldsetSelector)
+
+            const indcutoptswrap = setIndCutOptWrap(fileID, filename)
+
+            fieldset.append(indcutoptswrap)
+            if ($.type(fileContents) === 'string') {
+                j++
```

Comment:
Please, Please, don't do this. WTF is `j`... isn't it just the length? Can't you just get the length?